src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/utils/NodeProfUtil.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package ch.usi.inf.nodeprof.utils;
+
+import com.oracle.truffle.api.instrumentation.InstrumentableNode;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.js.nodes.access.GlobalScopeVarWrapperNode;
+import com.oracle.truffle.js.nodes.instrumentation.JSInputGeneratingNodeWrapper;
+import com.oracle.truffle.js.nodes.instrumentation.JSTaggedExecutionNode;
+
+public class NodeProfUtil {
+    public static Node getParentSkipWrappers(Node n) {
+        assert n != null;
+        Node parent = n.getParent();
+        while (parent != null && (isWrapperNode(parent) || parent.getSourceSection() == null)) {
```

Comment:
Maybe it is also needed to check if the parent node is generator wrapper as well: https://github.com/graalvm/graaljs/blob/master/graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/nodes/control/GeneratorWrapperNode.java

Reply:
Yes looks like it should be added. Maybe @eleinadani can confirm?

Reply:
Yes, you might have a generator wrapper wrapped by an instrumentation wrapper so one more extra check would not hurt -- you can have a look at the class hierarchy, since I think that generator wrapper inherits from the same common wrapper interface (or anyway an interface used by other wrappers e.g. generator expr blocks)

Reply:
@eleinadani `GeneratorWrapperNode` does not derive from the common wrapper interface, I suppose because it's not an instrumentation wrapper but a special purpose wrapper similar to e.g. `GlobalScopeVarWrapperNode`.

For our purposes here (skipping over nodes during `getParent`) I think it makes sense to treat it like a wrapper, so I'll make the change. We should however already skip it due to the `parent.getSourceSection() == null` check in `getParentSkipWrappers`.

Reply:
ah, yes -- you are right; it's indeed an "internal" wrapper used to resume both generators and async functions. Makes sense to consider it as a "normal" wrapper